### PR TITLE
[10-10EZ] Delete hca_submission_failure_mailer

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): NO
- Removing the old `hca_submission_failure_mailer` as we are now using VANotify to send this email. We will merge this after we turn the feature on in production. 
- 1010 Health Apps

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/85948
## Testing done

- [x] *New code is covered by unit tests*
- Previously it would use the old ApplicationMailer if the toggle was off. Now it will _always_ use VANotify to send an email.

## What areas of the site does it impact?
1010-EZ 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by Mitch-A6 — APPROVED** [review]

**Review by rmtolmach — DISMISSED** [review]

**Comment by bosawt** [comment]
@coope93 Do you know what file caused Identity to be tagged here? To be honest I don't think this PR should require an OCTO Identity approval

**Comment by bosawt** [comment]
> > @coope93 Do you know what file caused Identity to be tagged here? To be honest I don't think this PR should require an OCTO Identity approval
> 
> @bosawt I do not. All I can figure is the `app/sidekiq/transactional_email_analytics_job.rb`?

Oh, I found it, it's `spec/sidekiq/transactional_email_analytics_job_spec.r`. Can you change this from `onto-identity` to `backend-review-group` in CODEOWNERS?

**Review by bosawt — APPROVED** [review]
Approving the CODEOWNERS change to remove identity from the spec file

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**
